### PR TITLE
events by persistenceid improvements

### PR DESCRIPTION
**Pull request by WellingR**
This improved solution closes #241

Only the configured batch size of events are fetched each time. This implementation is based on the implementation of the eventsByTag query

### CODE REVIEW

**Comment by WellingR** [comment]
@renatocaval @richbirch please review

File: src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala [hunk 1]
```diff
@@ -123,36 +123,53 @@ class JdbcReadJournal(config: Config, configPath: String)(implicit val system: E
...
+      case (from, control) =>
+        def retrieveNextBatch(): Future[Option[((Long, FlowControl), Seq[PersistentRepr])]] = {
+          for {
+            xs <- currentJournalEventsByPersistenceId(persistenceId, from, toSequenceNr, batchSize).runWith(Sink.seq)
+          } yield {
+            val hasMoreEvents = xs.size == batchSize
+            // Events are ordered by sequence number, therefore the last one is the largest
+            val hasLastEvent = xs.lastOption.exists(last => last.sequenceNr >= toSequenceNr)
+            val nextControl: FlowControl =
+              if (hasLastEvent || from > toSequenceNr) Stop
```
Comment (renatocaval) [inline src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala]: If `hasLastEvent` is true, we should `ContinueDelayed`. This is expected to be an infinite stream of events. 

I think we will need another private method that we can use here in which we indicate if we want to stop on the last event or if we want to continue polling. I mean, `currentEventsByPersistenceId` should also benefit from this optimization. 

For `eventsByTag` we have `private def eventsByTag(tag: String, offset: Long, terminateAfterOffset: Option[Long])`. And we use `terminateAfterOffset` to decide if it should keep polling or not.

File: src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala [hunk 2]
```diff
@@ -123,36 +123,53 @@ class JdbcReadJournal(config: Config, configPath: String)(implicit val system: E
...
+
+    Source.unfoldAsync[(Long, FlowControl), Seq[PersistentRepr]]((Math.max(1, fromSequenceNr), Continue)) {
+      case (from, control) =>
+        def retrieveNextBatch(): Future[Option[((Long, FlowControl), Seq[PersistentRepr])]] = {
+          for {
+            xs <- currentJournalEventsByPersistenceId(persistenceId, from, toSequenceNr, batchSize).runWith(Sink.seq)
+          } yield {
+            val hasMoreEvents = xs.size == batchSize
+            // Events are ordered by sequence number, therefore the last one is the largest
+            val hasLastEvent = xs.lastOption.exists(last => last.sequenceNr >= toSequenceNr)
```
Comment (renatocaval) [inline src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala]: We could have an optimization here: 

```scala
val lastSeqNrInBatch: Option[PersistentRepr] = xs.lastOption.map(_.sequenceNr)
val hasLastEvent = lastSeqNrInBatch.exists(_ >= toSequenceNr)

... then later

val nextFrom: Long = 
  lastSeqNrInBatch match {
    // Continue querying from the last sequence number (the events are ordered)
    case Some(lastSeqNr) => lastSeqNr + 1
    case None =>  from
  }
```

As it is now, we are traversing `xs` twice. It's a tiny optimization, but helpful when `xs` is large.

File: src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala (hunk 1 above)
Comment (WellingR) [reply src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala]: The trait we are implementing see [EventsByPersistenceIdQuery.eventsByPersistenceId](https://doc.akka.io/api/akka/current/akka/persistence/query/scaladsl/EventsByPersistenceIdQuery.html#eventsByPersistenceId(persistenceId:String,fromSequenceNr:Long,toSequenceNr:Long):akka.stream.scaladsl.Source[akka.persistence.query.EventEnvelope,akka.NotUsed]) has the `toSequenceNr` as a mandatory parameter, it documents the following:
> You can retrieve a subset of all events by specifying fromSequenceNr and toSequenceNr or use 0L and Long.MaxValue respectively to retrieve all events.

I agree with the point that `currentEventsByPersistenceId` could also benefit from this optimization, this requires retrieving the current `toSequenceNr` first though.

File: src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala (hunk 2 above)
Comment (WellingR) [reply src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala]: Implemented this.
I am not sure how much is really matters since `xs` should be a vector, and we only look at the last element, but anyways this is could be slightly better.

File: project/ProjectAutoPlugin.scala [hunk 3]
```diff
@@ -106,8 +106,8 @@ object ProjectAutoPlugin extends AutoPlugin {
  )
 
   def determineMimaPreviousArtifacts(scalaBinVersion: String): Set[ModuleID] = {
-    val compatVersions: Set[String] = if (scalaBinVersion.startsWith("2.13")) Set.empty else {
-      Set("3.5.0")
+    val compatVersions: Set[String] = if (scalaBinVersion.startsWith("2.13")) Set("3.5.2") else {
+      Set("3.5.0", "3.5.1", "3.5.2")
```
Comment (renatocaval) [inline project/ProjectAutoPlugin.scala]: We can simplify it by only checking against 3.5.2. Since 3.5.0 and 3.5.1 are binary compatible with 3.5.2. 

This can be on another PR
